modelexpress_server/src/services.rs

```diff
@@ -174,6 +242,175 @@ impl ModelService for ModelServiceImpl {
+
+        // Spawn a task to stream files
+        tokio::spawn(async move {
+            // Allocate buffer once and reuse across all files
+            let mut buffer = vec![0u8; chunk_size];
+            let mut is_first_chunk = true;
+
+            for (file_idx, (relative_path, total_size)) in files.iter().enumerate() {
+                let file_path = model_path.join(relative_path);
+                let is_last_file = file_idx == total_files.saturating_sub(1);
+
+                debug!("Streaming file: {:?} ({} bytes)", relative_path, total_size);
+
+                // Open the file
+                let file = match tokio::fs::File::open(&file_path).await {
+                    Ok(f) => f,
+                    Err(e) => {
+                        error!("Failed to open file {:?}: {}", file_path, e);
+                        let _ = tx
+                            .send(Err(Status::internal(format!("Failed to open file: {e}"))))
+                            .await;
+                        return;
+                    }
+                };
+
+                let mut reader = tokio::io::BufReader::new(file);
+                let mut offset: u64 = 0;
+
+                loop {
+                    let bytes_read = match reader.read(&mut buffer).await {
+                        Ok(0) => break, // EOF
+                        Ok(n) => n,
+                        Err(e) => {
+                            error!("Failed to read file {:?}: {}", file_path, e);
+                            let _ = tx
+                                .send(Err(Status::internal(format!("Failed to read file: {e}"))))
+                                .await;
+                            return;
+                        }
+                    };
+
+                    let is_last_chunk = offset.saturating_add(bytes_read as u64) >= *total_size;
+
+                    let chunk = FileChunk {
+                        relative_path: relative_path.to_string_lossy().to_string(),
+                        data: buffer[..bytes_read].to_vec(),
+                        offset,
+                        total_size: *total_size,
+                        is_last_chunk,
+                        is_last_file: is_last_file && is_last_chunk,
+                        commit_hash: if is_first_chunk {
+                            is_first_chunk = false;
+                            commit_hash.clone()
+                        } else {
+                            None
+                        },
+                    };
+
+                    if tx.send(Ok(chunk)).await.is_err() {
+                        debug!("Client disconnected during file stream");
+                        return;
+                    }
+
+                    offset = offset.saturating_add(bytes_read as u64);
+                }
+            }
+
+            info!("File streaming completed for model");
+        });
+
+        Ok(Response::new(ReceiverStream::new(rx)))
+    }
+
+    async fn list_model_files(
```

Comment:
I don't see this is being called anywhere. Is this needed for this PR?

Reply:
It may have ended up as dead code. I'll double check this.